### PR TITLE
Set permissions on JSON files

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Set permissions for saved JSON files to 600.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: tests/utils/test_json.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""test json."""
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from supervisor.utils.json import write_json_file
+
+
+def test_file_permissions():
+    """Test file permissions."""
+    with TemporaryDirectory() as temp_dir:
```
Comment (MartinHjelmare) [inline tests/utils/test_json.py:10]: Pytest has a `tmp_path` fixture.
https://docs.pytest.org/en/stable/tmpdir.html

Comment (ludeeus) [reply tests/utils/test_json.py:10]: Correct, but we are not using that anywhere else?

File: tests/utils/test_json.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+"""test json."""
+from pathlib import Path
+
+from supervisor.utils.json import write_json_file
+
+
+def test_file_permissions(tmp_path):
+    """Test file permissions."""
+    tempfile = Path(tmp_path, "test.json")
```
Comment (MartinHjelmare) [inline tests/utils/test_json.py:9]:
```suggestion
    tempfile = tmp_path / "test.json"
```

File: tests/utils/test_tarfile.py [hunk 3]
```diff
@@ -70,72 +69,80 @@ def test_is_exclude_by_filter_bad():
...
-        with SecureTarFile(temp_tar, "r", key=key) as tar_file:
-            tar_file.extractall(path=temp_new, members=tar_file)
-
-        assert temp_new.is_dir()
-        assert temp_new.joinpath("test_symlink").is_symlink()
-        assert temp_new.joinpath("test1").is_dir()
-        assert temp_new.joinpath("test1/script.sh").is_file()
-        assert temp_new.joinpath("test1/script.sh").stat().st_mode == 33261
-        assert temp_new.joinpath("README.md").is_file()
+    temp = Path(tmp_path)
```
Comment (MartinHjelmare) [inline tests/utils/test_tarfile.py:113]: This isn't needed.

**Review by pvizeli — APPROVED** [review]

**Comment by ludeeus** [comment]
image
For 1 added line to the code :see_no_evil:

**Pull request merged**